### ISSUE TITLE
Offer additional information/hints whereever price distance in % (dynamic pricing) is used

### ISSUE THREAD

**Issue by ripcurlx** [bisq-network/bisq#1214]
There are multiple places in the app where the a percentage number is shown next to the price or can be entered. Although we display e.g. in the create offer screen that you can enter the "Distance in % from the market price" it is not explained if this will adapt if the bitcoin price is changing. Also the user might not be 100% sure if the BitcoinAverage price will be used or another price feed. One solution could be to add a (i) icon next to the input field displaying this information. The same can be done in tables where we display the price e.g. 8020 (4%) --> add an (i) icon next to the percentage explaining the reasoning behind this number.

**Comment by ripcurlx** [comment]
If we render a bar underneath the price itself for the price distance/spread it could look like this. Red would mean price distance is negative for the offer taker and green means it is positive for the offer taker. 
![sell screen](https://user-images.githubusercontent.com/170962/35928468-83e87136-0c2d-11e8-8b47-3ab3c79c6179.png)

**Comment by ripcurlx** [comment]
Of course most of the time it will be red underlays and no underlay if the price distance is 0%

**Comment by ripcurlx** [comment]
See https://github.com/bisq-network/exchange/issues/1203#issuecomment-368877577 how this could be done for the create offer screen.

**Comment by GuiSim** [comment]
What would the content of the (i) "popup" be? How would you explain this difference.

**Comment by ripcurlx** [comment]
@GuiSim Did you see my comment from the other issue https://github.com/bisq-network/exchange/issues/1203#issuecomment-368877577? It wouldn't be an (i) icon, but just the sync icon and on roll over we have enough space to explain the exact details on this dynamic price distance offer.
![sell screen no flags](https://user-images.githubusercontent.com/170962/36736195-4b6ff120-1bd8-11e8-8366-f5a4fd580d6c.png)

**Comment by GuiSim** [comment]
I had missed that. Would you prefer to have comments on this tooltip here?

I still feel like it's not super clear what "below the market price" really entails. I had written some comments in https://github.com/bisq-network/exchange/issues/1191#issue-288180241 but essentially I wonder if the text should be different if you're buying VS selling.

I would also suggest that instead of saying "and updated every minute" it could say "This offer is 2% below the market price as of `X seconds ago`." or similar. This would make it more obvious when was the last update and also would make it more apparent that the sync button worked.

"Updated every minute" could be a setting or explained elsewhere, it doesn't have to be on every line.

Overall I think this proposal is a great step in the right direction. I hope my feedback can be useful! 
What do you think?

**Comment by ripcurlx** [comment]
Yes, to update the text live might be a good way to even indicate when exactly the price changed.
We can and should have different descriptions for buy and sell offers. Now we have the space to make it absolutely clear to the user.

Maybe for "Sell BTC":
This offer is 2% below the market price (last update 20 seconds ago). You are getting less than the market price when taking this offer.

"Buy BTC":
This offer is 2% below the market price (last update 20 seconds ago). You are paying less than the market price when taking this offer.

**Comment by ripcurlx** [comment]
We could also use the sync icon within the create offer screen and re-order the fields a little bit. With the green (i) or orange warning we could show in the tooltip what this percentage means for the buyer/seller.
![create offer](https://user-images.githubusercontent.com/170962/36739146-85caa2d2-1bdf-11e8-940f-b9e31da48440.png)
![create offer with warning](https://user-images.githubusercontent.com/170962/36739156-8df3918a-1bdf-11e8-9509-25cfe286c90b.png)

**Comment by GuiSim** [comment]
Very good ideas 👍

**Comment by ripcurlx** [comment]
The roll-over popups for creating a buy offer could say something like:
**Buy below market price (2% distance)** 
![buy below market price](https://user-images.githubusercontent.com/170962/36779181-7598cc96-1c6f-11e8-807c-d808144b98da.png)
**Buy at market price (0% distance)** 
![buy at market price](https://user-images.githubusercontent.com/170962/36779183-75b9e660-1c6f-11e8-9840-40bbc31e915c.png)
**Buy above market price (-2% distance)** 
![buy above market price](https://user-images.githubusercontent.com/170962/36779184-75d9ec3a-1c6f-11e8-98ee-1209670dec97.png)

**Comment by ripcurlx** [comment]
And the same roll-over popups for creating a sell offer:
**Sell above market price (2% distance)** 
![create sell offer plus 2 percent](https://user-images.githubusercontent.com/170962/36782197-7962d4da-1c78-11e8-8898-b2fa1141205f.png)
**Sell at market price (0% distance)** 
![create sell offer zero percentage](https://user-images.githubusercontent.com/170962/36782202-7f4e072a-1c78-11e8-8798-dcbad273022f.png)
**Sell below market price (-2% distance)** 
![create sell offer minus 2 percent](https://user-images.githubusercontent.com/170962/36782213-846284f2-1c78-11e8-9edd-9e7beb14535e.png)

**Comment by ripcurlx** [comment]
For fixed offers I don't think we need additional information. Similar to the percentage scenario I just disabled the input field below, so it's not so confusing where you need/should change values. What do you think?
![create offer fixed](https://user-images.githubusercontent.com/170962/36782818-74cfa374-1c7a-11e8-8c1a-28d69d80d3b9.png)

**Comment by tr37ion** [comment]
Good ideas. Can you add a bit more space here?

![image](https://user-images.githubusercontent.com/1538617/36790386-87a00b44-1c8c-11e8-8aff-3314f85ef523.png)

**Comment by ManfredKarrer** [comment]
Not sure if putting the tab button above is the best place as the text field headline is in between and makes the visual connection of the tab button with the below text fields less clear.
Maybe we could make the icons for "Percentage" and "Fixed" a part of the text input+currency combo fields and add a swap button in between on the right so the user can swap between the 2 options? The "Percentage" and "Fixed" text can be part of the label description (is partly already). The not active (2nd row) label description cold be also greyed out so it becomes more clear that its not active part.
Maybe the % icon can be a combination of the BitcoinAverage/Poloniex and something liker here: https://www.iconfinder.com/icons/189569/label_percentage_price_sale_shop_shopping_tag_icon#size=512 (probably too large)?
Though I also fear that the BitcoinAverage/Poloniex icon are too big and attract to much attention as they are colored....

**Comment by ripcurlx** [comment]
The reason why I've put it on top is to have more connection with the textfields ;-). We could even colorize the section below, but actually I like the idea of getting rid of them completely. I'll create a mockup to see, if it might be enough to work with placeholder text and icons next to the input fields. I'm not sure if it make sense to add the BitcoinAverage/Poloniex icon into the mix as I guess hardly anyone would recognize these logos.

**Comment by ripcurlx** [comment]
One more thing I just recognized. Why do we have an amount range (min, max) on the left side, but no amount range in USD on the right side? Isn‘t that confusing that we are not  communicating the range you might have to pay there as well? In general I want to get rid of the range selection by default anyways, but atm I think it‘s not very consistent. I‘ll post new screens tomorrow morning.

**Comment by ManfredKarrer** [comment]
@ripcurlx Yes agree that BitcoinAverage/Poloniex icons will probably not be a good idea, also we might create our own price mix one day...
To use the placeholder text for explaining what % and fixed is is a good idea. Its partly already but text could be improved.
 Re range: Yes agree. If you find a way how to do it without overloading the UI would be good. Though as nobody to my knowledge requested that and also myself never thought on that, I am not sure if its really a missing feature?
Maybe the range could be replaced by a more visual component like a progressbar where u can set start and end range. That would be limited automatically by the trade limits. Of course text fields are required as well as some want to edit it exactly to a specific number. But such a component might be then also used on the counter currency side (fiat/altcoin).

**Comment by ripcurlx** [comment]
I just removed the range by default for now (will improved with selector soon) and played around with the idea not having the tabs at all. We could think about using radio buttons for selection instead. What do you think?
![create sell offer reduced](https://user-images.githubusercontent.com/170962/36896476-276ab64c-1e13-11e8-8941-4067f592324c.png)

**Comment by ripcurlx** [comment]
Or without RadioButtons like that:
![create sell offer reduced](https://user-images.githubusercontent.com/170962/37027095-b1e6386c-2130-11e8-8bc9-e1b917c51eb4.png)

**Comment by ripcurlx** [comment]
And if you want to add a minimum amount it could look like this:
![create sell offer reduced range](https://user-images.githubusercontent.com/170962/37027525-d44d1762-2131-11e8-81d4-bc8d7dbf1fc0.png)

**Comment by ripcurlx** [comment]
Changed the style of "add min. amount" a little bit and added a tooltip how the information for this range feature could look like.
![create sell offer reduced](https://user-images.githubusercontent.com/170962/37082613-44335c74-21ed-11e8-8896-e7dcbc33a45f.png)

**Comment by ManfredKarrer** [comment]
I would phrase it more like that:
"Defining a minimum amount increases the chance your offer gets taken faster as the taker can choose any amount within the given range."
If you use buyer ore seller then u need to use opposite of what user is doing, e.g. here its the seller. I think taker might be less confusing...

**Comment by ripcurlx** [comment]
Good suggestion :+1: - so it works for both scenarios.

**Comment by ripcurlx** [comment]
@GuiSim and @tr37ion: How do you like https://github.com/bisq-network/exchange/issues/1214#issuecomment-371068146 and https://github.com/bisq-network/exchange/issues/1214#issuecomment-370736910 in the create offer screen? I slowly want to move this issue into a ready-for-development state, so it will be part of v0.7.0.

**Comment by tr37ion** [comment]
To me https://github.com/bisq-network/exchange/issues/1214#issuecomment-369894685 is the most appealing as the cross marks the choice the best. Adding those info tooltips is always a plus.

**Comment by GuiSim** [comment]
@tr37ion if I understand this correctly, the 2nd screenshot is just the result of clicking the `+` in the first screenshot. 

I think it's _okay_ but I think it's somewhat confusing. I didn't spend much time focusing on this part of the application (I mostly thought about how the buy/sell table was displayed).

I think that to me it would make more sense if the flow was like this:

Do you want to sell an exact amount of BTC or a range?

If it's exact, prompt only for one amount.
If it's a range, prompt for a min and a max.

I know this is challenging what has been discussed so far and I'm sorry this feedback comes in late. Though to be honest I didn't plan to participate in this discussion until you pinged me ;) 

If it's too far off and not relevant, please ignore. I don't want to waste your time with this, just my 2¢.

**Comment by ManfredKarrer** [comment]
@ripcurlx 
I basically like the last screen graphically but I am not sure if hiding the min amount is really the best way. See https://github.com/bisq-network/exchange/issues/1444 for the reason of my concerns...

Maybe a slider component might work to satisfy the issue? If the user set min. amount above 0.00625 BTC we could show a popup telling the background that new users has that limit and if he set min. amount higher he excluded those users.

**Comment by tr37ion** [comment]
A slider like this one?

![image](https://user-images.githubusercontent.com/1538617/37247544-78bc9b80-24b4-11e8-9584-9e66b6745ae8.png)

**Comment by ripcurlx** [comment]
@GuiSim It's never too late for feedback 😉 
Reading through all the suggestions I think we should split this issue up further to prevent a gigantic change in this view. I want to give the min/max amount combined with the trade-limit issue some time to work it out properly and start implementing the fixed/dynamic part first. Still I want to hide the min amount by default, as it was super confusing to me when I saw it the first time. Also as I didn't understand it seeing it for the first time, I didn't use it on my first offer. The additional advantage of hiding it is, that we have the chance to communicate in more detail in the pop over why it makes sense to set an amount range.

General Changes compared to current  version:
- New selection for fixed/dynamic price

Changes for selling fixed amount:
- Label "Fixed amount of BTC to sell" instead of "Amount of BTC to sell"
- Hide Minimum amount by default (@tr37ion I used the plus circle as we already use the circle style (e.g. minus) on other places within the app)
![create sell offer fixed amount](https://user-images.githubusercontent.com/170962/37590179-c4f6a0d6-2b67-11e8-9599-66c1e6aeec1e.png)

Changes for selling amount range:
- Label "Maximum amount of BTC" instead of "Amount of BTC to sell"
- Added Minimum amount of BTC that can be removed again by clicking on the minus circle icon
- Label "Maximum amount in USD" instead of "Amount in USD to receive"
- Added Minimum amount in USD

I've added the minimum and maximum wording in this view to make it more consistent with the min-max used on other places within the app.

![create sell offer amount range](https://user-images.githubusercontent.com/170962/37589742-72b1e0ac-2b66-11e8-836c-6c52381878b1.png)

**Comment by ManfredKarrer** [comment]
@ripcurlx Graphically it looks good to me. Only small suggestion is to move the blue swap button up to the middle of the 2 rows. But up to you...
I still have concerns if the hiding of the min. amount field has negative effects as offers without a lower min amount might not be accessible for new users with the low 0.0625 BTC limit in the first month.

What if we suggest by default a min. of 0.0625 (e.g. the min. for that payment account) and if the user increases that he gets a one time popup where the reasoning behind is explained (fewer potential takers). We still can keep the +/- buttons but just use default the state where the min. amount is visible and filled with the payment accounts limit for the first month (25% of payment method limit).
The state change the user is doing (e.g. remove the min amount or set it higher or lower could be persisted in the preferences and next time get applied so we don't bother him if he does not follow the suggestion.

What do you think?

**Comment by ripcurlx** [comment]
I do understand your concerns regarding the minimum amount issue, that might exclude new traders from the opportunity to take most of the available offers. I thought there might be an intermediate step before we've to make bigger changes within this screen, but I think we need to fix this properly in one step.

Until we'll have a good solution for the amount/limit problem, I'll make the changes for Fixed|Percentage first. I've put the swap sign at the bottom field, as I remembered it like that in Google Maps, but after revisiting it I saw that they just put it there as they needed an icon next to the field above as well.
So the next iteration will look like that:
![create sell offer market price margin](https://user-images.githubusercontent.com/170962/37896584-671efdfa-30e4-11e8-95e1-de05f82acae1.png)

**Comment by ManfredKarrer** [comment]
Here is my suggestion for the min amount handling in more details:
We could use a preferences checkbox for "Set min. amount to  not exclude new users with min. trade limit". If that flag is set (default its true) then the min. amount is always the 0.25% of the payment methods trade limit. If the user changes the min. amount the first time he gets a popup explaining the context and asking if he wants to change that flag and if he wants to keep min amount to max amount we remember that and show the + icon and don't show the min amount fields. If he changes it the min. amount he entered will be persisted and applied to new offers. E.g. if he sets it to 0.01 his next offer will have set it automatically to 0.01 as well. If he sets it to max. the next offer will have it to max again. 

So to summarize it will require 3 new preference entries:
- boolean useMinLimit
- boolean dontUseMinAmount
- long preferredMinAmount

The useMinLimit is reflected as checkbock in general preferences.
dontUseMinAmount is set to true if user set min amount to max amount (in first time popup the user can set that in the dialoge).
If min amount is set the preferredMinAmount is set. The preferredMinAmount must be checked with the amount field as it is now to not have larger preferredMinAmount than amount. But we only persist changes of preferredMinAmount which have been entered by the user not which have been set by the app from input validation.

**Comment by ripcurlx** [comment]
I decided in the first iteration against having the exchange arrows in the middle and adding the extra space before the equal sign as I liked it better that way. But of course up for discussions. For trying it out please see https://github.com/bisq-network/bisq-desktop/pull/1477
